onnxruntime/core/providers/cpu/controlflow/loop.cc

```diff
@@ -494,7 +494,7 @@ Status LoopImpl::Execute(const FeedsFetchesManager& ffm) {
     status = utils::ExecuteSubgraph(session_state_, ffm, feeds, fetches, {},
                                     ExecutionMode::ORT_SEQUENTIAL, context_.GetTerminateFlag(), context_.Logger(),
                                     context_.GetComputeStream());
-
+    context_.GetComputeStream()->Flush();
```

Comment:
please add a comment here why we need flush. it mainly because the condition needs to be accessed on CPU.